common/djangoapps/util/enterprise_helpers.py

```diff
@@ -120,3 +122,50 @@ def insert_enterprise_pipeline_elements(pipeline):
 
     for index, element in enumerate(additional_elements):
         pipeline.insert(insert_point + index, element)
+
+
+def get_enterprise_customer_logo_url(request):
+    """
+    Client API operation adapter/wrapper
+    """
+    logo_url = None
+
+    if not enterprise_enabled():
+        return None
+
+    parameter = get_enterprise_branding_filter_param(request)
+    if parameter:
+        provider_id = parameter.get('provider_id', None)
+        ec_uuid = parameter.get('ec_uuid', None)
+
+        branding_info = enterprise_api.get_enterprise_branding_info(provider_id=provider_id, ec_uuid=ec_uuid)
+        if branding_info and branding_info.logo:
+            logo_url = branding_info.logo.url
+
+    return logo_url
+
+
+def set_enterprise_branding_filter_param(request, provider_id):
+    """
+    setting 'EC_BRANDING_FILTER_PARAM' in session.
+    'EC_BRANDING_FILTER_PARAM' either be provider_id or ec_uuid
+    e.g. {provider_id: 'xyz'} or {ec_src: enterprise_customer_uuid}
+    """
+
+    if EC_BRANDING_FILTER_PARAM not in request.session:
+        # SSO based Enterprise customers provide SAML idp e.g. provider_id
+        if provider_id:
+            request.session[EC_BRANDING_FILTER_PARAM] = {'provider_id': provider_id}
+
+        elif request.GET.get('ec_src', None):
+            # we are assuming that none sso based enterprise will return Enterprise Customer uuid as 'ec_src' in query
+            # param e.g. edx.org/foo/bar?ec_src=6185ed46-68a4-45d6-8367-96c0bf70d1a6
+            request.session[EC_BRANDING_FILTER_PARAM] = {'ec_uuid': request.GET['ec_src']}
+
+
+def get_enterprise_branding_filter_param(request):
+    """
+    :return Filter parameters from session for enterprise customer branding information.
+
+    """
+    return request.session.get(EC_BRANDING_FILTER_PARAM, None)
```

Comment:
By this, user will get his/her enterprise logo on logistration pages as well as on his/her dashboard after successful login. session will be destroyed on user logout.